packages/dashboard-backend/src/api/dockerConfigApi.ts

```diff
@@ -0,0 +1,139 @@
+    `${baseApiPath}/namespace/:namespace/dockerconfigs`,
+    getSchema({ tags, params: namespacedDockerConfigSchema, body: dockerConfigSchema }),
+    async function (request: FastifyRequest) {
+      const { namespace } = request.params as restParams.INamespacedParam;
+      const { dockerconfig } = request.body as restParams.ISecreteParams;
+      const { secretApi } = await getDevWorkspaceClient(request);
+      const body = createDockerConfigSecret(dockerconfig);
+      const secret = await secretApi.create(namespace, body);
+      return createResponse(secret);
+    }
+  );
+
+  server.put(
+    `${baseApiPath}/namespace/:namespace/dockerconfig`,
+    getSchema({ tags, params: namespacedDockerConfigSchema, body: dockerConfigSchema }),
+    async function (request: FastifyRequest) {
+      const { namespace } = request.params as restParams.INamespacedParam;
+      const { dockerconfig, resourceVersion } = request.body as restParams.ISecreteParams;
+      const { secretApi } = await getDevWorkspaceClient(request);
+      let secret: V1Secret | undefined;
+      try {
+         secret = await secretApi.read(namespace, secretName);
+      } catch (e) {
+         if ((e as { statusCode?: number } | undefined)?.statusCode === 404) {
+           const body = createDockerConfigSecret(dockerconfig);
+           const resp = await secretApi.create(namespace, body);
+           return createResponse(resp);
+         }
+        throw e;
+      }
+      updateDockerConfigSecret(secret, dockerconfig, resourceVersion);
+      const resp = await secretApi.replace(namespace, secretName, secret);
+      return createResponse(resp);
+    }
+  );
+
+  server.get(
+    `${baseApiPath}/namespace/:namespace/dockerconfig`,
+    getSchema({ tags, params: namespacedDockerConfigSchema }),
+    async function (request: FastifyRequest) {
+      const { namespace } = request.params as restParams.INamespacedParam;
+      const { secretApi } = await getDevWorkspaceClient(request);
+      let secret: V1Secret | undefined;
+      try {
+        secret = await secretApi.read(namespace, secretName);
+      } catch (e) {
+        if ((e as { statusCode?: number } | undefined)?.statusCode === 404) {
+          return createResponse(secret);
+        }
+        throw e;
+      }
+      return createResponse(secret);
+    }
+  );
+
+  server.delete(
+    `${baseApiPath}/namespace/:namespace/dockerconfig`,
+    getSchema({ tags, params: namespacedDockerConfigSchema }),
+    async function (request: FastifyRequest, reply: FastifyReply) {
+      const { namespace } = request.params as restParams.INamespacedParam;
+      const { secretApi } = await getDevWorkspaceClient(request);
+      const { status, message } = await secretApi.delete(namespace, secretName);
+      if (status === 'Failure') {
+        throw new Error(message);
+      }
+      reply.code(204);
+      return reply.send();
+    }
+  );
+}
+
+function createDockerConfigSecret(dockerconfig: string): V1Secret {
```

Comment:
I would say that's not creation but conversion 
```suggestion
function toDockerConfigSecret(dockerconfig: string): V1Secret {
```